dali/operators/image/crop/bbox_crop.cc

```diff
@@ -14,143 +14,635 @@
+  bool Setup(std::vector<OutputDesc> &output_desc, const workspace_t<CPUBackend> &ws) {
+    if (bbox_layout_.empty()) {
+      auto ltrb = spec_.GetArgument<bool>("ltrb");
+      bbox_layout_ = ltrb ? DefaultLayout<ndim>() : DefaultAnchorAndShapeLayout<ndim>();
+    }
+
+    auto default_bbox_layout_start_end = DefaultLayout<ndim>();
+    auto default_bbox_layout_start_shape = DefaultAnchorAndShapeLayout<ndim>();
+    DALI_ENFORCE(bbox_layout_.is_permutation_of(default_bbox_layout_start_end) ||
+                 bbox_layout_.is_permutation_of(default_bbox_layout_start_shape),
+      make_string("`bbox_layout` should be a permutation of `", default_bbox_layout_start_end,
+                  "` or `", default_bbox_layout_start_shape, "`. Got: `", bbox_layout_, "`"));
+
+    if (spec_.ArgumentDefined("input_shape") && spec_.ArgumentDefined("crop_shape")) {
+      CollectShape(input_shape_, "input_shape", spec_, ws, ndim, true);
+      CollectShape(crop_shape_, "crop_shape", spec_, ws, ndim, true);
+
+      // Converting the shapes to "WHD" or "WH" if necessary
+      auto default_shape_layout = InternalShapeLayout(ndim);
+      if (!shape_layout_.empty() && shape_layout_ != default_shape_layout) {
+        DALI_ENFORCE(shape_layout_.is_permutation_of(default_shape_layout),
+                     make_string("`shape_layout` should be a permutation of ", default_shape_layout,
+                                 "` for the provided inputs"));
+        auto perm = GetDimIndices(shape_layout_, default_shape_layout);
+        for (int sample = 0; sample < static_cast<int>(input_shape_.size()); sample++) {
+          TensorShape<> in_shape = input_shape_[sample];
+          TensorShape<> crop_shape = crop_shape_[sample];
+
+          DALI_ENFORCE(crop_shape.sample_dim() == ndim,
+                    make_string("Unexpected number of dimensions. Expected ", ndim, ", got ",
+                                crop_shape.sample_dim()));
+
+          for (int i = 0; i < ndim; i++) {
+            int axis = perm[i];
+            DALI_ENFORCE(
+                crop_shape[axis] <= in_shape[axis],
+                make_string("Crop shape can't exceed input shape dimensions. Got crop_shape=",
+                            crop_shape, ", input_shape=", in_shape));
+            input_shape_[sample][i] = in_shape[axis];
+            crop_shape_[sample][i]  = crop_shape[axis];
+          }
+        }
+      }
+    }
+    return false;
   }
-}
 
-template <>
-void RandomBBoxCrop<CPUBackend>::RunImpl(SampleWorkspace &ws) {
-  const auto &boxes_tensor = ws.Input<CPUBackend>(0);
+  void Run(SampleWorkspace &ws) {
+    const auto &boxes_tensor = ws.Input<CPUBackend>(0);
+    auto nboxes = boxes_tensor.dim(0);
+    auto ncoords = ndim * 2;
+
+    std::vector<Box<ndim, float>> bounding_boxes(nboxes);
+    ReadBoxes(make_span(bounding_boxes),
+              make_cspan(boxes_tensor.data<float>(), boxes_tensor.size()), bbox_layout_);
+
+    std::vector<int> labels;
```

Comment:
Can't this be a `span` or a `TensorView`? It's a plain copy from the input and a needless allocation.